packages/adapter-vercel/index.js

```diff
@@ -390,6 +390,25 @@ const plugin = function (defaults = {}) {
 				);
 			}
 
+			// optional chaining to support older versions that don't have this setting yet
+			if (builder.config.kit.router?.resolution === 'server') {
+				// Create a separate edge function just for server-side route resolution.
+				// By omitting all routes we're ensuring it's small (the routes will still be available
+				// to the route resolution, becaue it does not rely on the server routing manifest)
+				await generate_edge_function(
+					`${builder.config.kit.appDir}/routes`,
+					{
+						...defaults,
```

Comment:
should we add a `regions: 'all'` here to override any user options? The developer could have specified a region near their backend, but presumably there's no reason for the routing function to be constrained by that

Reply:
That seems right (omitting defaults), yeah

> Also, if we were to enhance `reroute` with the ability to do async then you _could_ fetch stuff from your backend.

I thought we'd decided against doing that because it's a footgun? I think we're better off keeping it sync

Reply:
Oh, I mistook the question in https://github.com/sveltejs/kit/pull/13379#pullrequestreview-2581459252 as meaning to make them async. I'm not sure how useful cookie/header support but no async really is. If you store your flags for A/B testing encrypted in some way you probably also request a third party service or edge DB to get the flag values.

Reply:
Your A/B testing middleware could do that lookup and chuck in a header with the result for the benefit of `reroute`

Reply:
If I use middleware I would also use it to do the reroute right then and there instead of spreading it out across multiple points.

Reply:
Be that as it may, it seems tangential to the question at hand. I'll remove `...defaults` and pass through `external`